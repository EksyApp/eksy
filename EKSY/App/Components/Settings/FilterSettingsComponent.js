import React, {Component} from 'react'
import {View, StyleSheet} from 'react-native'
import CheckBoxList from "../Common/CheckBoxList";
import Divider from "../Common/Divider";
import Label from "../Common/Label";
import PropTypes from 'prop-types'
import {FiltersShape} from "../../Utils/PropTypeShapes";

export default class FilterSettingsComponent extends Component {
	
	render() {
		return(
				<View>
					<Divider style={styles.divider} />
					<Label>Filters</Label>
					<CheckBoxList data={this.props.data} onPress={(name, checked) => {this.props.onPress(name, checked)}} />
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	divider: {
		marginTop: 30
	}
})

FilterSettingsComponent.propTypes = {
	data: FiltersShape,
	onPress: PropTypes.func
}