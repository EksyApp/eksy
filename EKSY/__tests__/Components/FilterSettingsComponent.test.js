import 'react-native';
import React from 'react';
import FilterSettingsComponent from '../../App/Components/Settings/FilterSettingsComponent';
import renderer from 'react-test-renderer';

test('FilterSettingsComponent renders correctly', () => {
	const tree = renderer.create(
			<FilterSettingsComponent data = {[{
				name: "Violence",
				stateDescription: "Show violent content"
			}]}/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});
