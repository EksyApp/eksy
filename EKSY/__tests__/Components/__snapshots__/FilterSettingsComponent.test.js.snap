// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSettingsComponent renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e1e8ee",
          "height": 1,
        },
        Array [
          Object {
            "backgroundColor": "rgb(112,112,112)",
            "height": 1,
            "margin": 3,
            "marginLeft": 10,
            "marginRight": 10,
          },
          Object {
            "marginTop": 30,
          },
        ],
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "rgb(101,114,122)",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Filters
  </Text>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 1,
      }
    }
    renderRow={[Function]}
    renderScrollComponent={[Function]}
    style={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(226,227,217)",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              false,
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            onLongPress={undefined}
            onPress={undefined}
            style={
              Array [
                Object {
                  "color": "#bfbfbf",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ï‚–
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "#43484d",
                    "fontWeight": "bold",
                    "marginLeft": 10,
                    "marginRight": 10,
                  },
                  Array [
                    Object {
                      "color": "rgb(89,85,78)",
                    },
                    undefined,
                  ],
                  undefined,
                ],
              ]
            }
          >
            Show violent content
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
